File: Data Structure/src/go/sorting/insertion_sort.go
package main
import "fmt"

func insertion_sort(arr []int, n int){
	for i := 0; i < n; i++{
		j := i
		for j>0 && arr[j-1] > arr[j] {
			var temp int = arr[j-1]
			arr[j-1] = arr[j]
			arr[j] = temp
			j--
		}

	}
}

func main() {
	arr := []int8{22, 11, 33, 44, 66, 55} // Inferred length
	var n int = len(arr)
	insertion_sort(arr, n)
	fmt.Println(arr)
}

File: Data Structure/src/cpp/sorting/insertion_sort.cpp
#include <iostream>

using namespace std;

void insertion_sort(int arr[], int n)
{
    for (int i = 0; i < n; i++)
    {
        int j = i;
        while (j > 0 && arr[j - 1] > arr[j])
        {
            int temp = arr[j - 1];
            arr[j - 1] = arr[j];
            arr[j] = temp;
            j--;
        }
    }
}

int main()
{
    int arr[] = {22, 11, 44, 33};
    int n = sizeof(arr) / sizeof(arr[0]);

    insertion_sort(arr, n);
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    cout << endl;
}
